% this is a simplified version of 
% https://github.com/yihui/knitr/blob/master/inst/examples/knitr-beamer.Rnw
\documentclass{beamer}
\ifdefined\knitrout
  \renewenvironment{knitrout}{\begin{footnotesize}}{\end{footnotesize}}
\else
\fi
\usepackage{xcolor,colortbl}
\begin{document}

%%http://www.charlesapple.com/2014/04/two-recent-infographic-fails-you-ought-to-know-about/

\pgfdeclareimage[height=0.5cm]{university-logo}{images/CRUK_CAMBRIDGE_I_Pos_RGB_300}
\logo{\pgfuseimage{university-logo}}


\title{Introduction to Statistical analysis}
\author{Mark Dunning. Materials by Deepak Parashar and Sarah Dawson}
\date{2nd September 2014}
\institute[Bioinformatics Core] % (optional, but mostly needed)
{

  Cancer Research Uk\\
  Cambridge Research Institute\\
  Robinson Way\\
  Cambridge \\
}

% very important to use option [fragile] for frames containing code output!

{
\usebackgroundtemplate{\includegraphics[width=\paperwidth]{images/criBackdrop.jpg}}
\begin{frame}
  \titlepage
\end{frame}
}

\begin{frame}{Outline}
  \tableofcontents
  % You might wish to add the option [pausesections]
\end{frame}

\section{Introduction}
\begin{frame}{The point of statistics}
\begin{itemize}
\item{Rarely feasible to study the whole population that we are interested in, so we take a sample instead}
\item{Assume that data collected represents a larger population}
\item{Use sample data to make conclusions about the overall population}
\end{itemize}
\includegraphics[height=5cm]{images/aSample.png}

\end{frame}

\begin{frame}{Data}
\begin{itemize}
\item{Type?}
\begin{itemize}
\item{Categorical (nominal) , e.g. Gender}
\item{Categorical with ordering (ordinal), e.g. Tumour grade }
\item{Discrete, e.g. Shoe size}
\item{Continuous, e.g. Body weight in kg}
\end{itemize}
\item{Independent  or dependent measurements}
\item{Representative of which population?}
\item{Distribution}
\begin{itemize}
\item{Normally distributed? Skewed? Bimodal?}
\end{itemize}
\end{itemize}
\end{frame}

\begin{frame}{Data type - example}
\begin{itemize}
\item{Sucess / failure of achieving a taks for a mouse which may be wild-type of knock-out, male or female, 2, 4 or 6 months old}
\end{itemize}
\includegraphics[height=5cm]{images/dataTypes1.png}
\end{frame}

\begin{frame}{Data type - example}
\begin{itemize}
\item{Gene expression in each cell sample which may be one of five cell-types (A,B,C,D,E)}
\end{itemize}
\includegraphics[height=5cm]{images/dataTypes2.png}
\end{frame}

\begin{frame}{Data type - example}
\begin{itemize}
\item{The number of bacteria for each subject which may be a cancer patient or a normal}
\end{itemize}
\includegraphics[height=5cm]{images/dataTypes3.png}
\end{frame}


\begin{frame}{Mesurements: Dependent / Independent?}
\begin{itemize}
\item{Measurements of gene expression taken from each of 20 individuals}
\item{Are any measurements more closely related than others?}
\begin{itemize}
\item{Siblings / littermates?}
\item{Same individual measured twice?}
\item{Batch effects?}
\end{itemize}
\item{If no reason - independent observations}
\end{itemize}
\end{frame}

\begin{frame}{Mesurements: Dependent / Independent?}
\begin{itemize}
\item{18 measurements; from repeating an experiment 6 times, each time in triplicate}
\end{itemize}
\includegraphics[height=5cm]{images/18measures.png}
\end{frame}

\begin{frame}{Mesurements: Dependent / Independent?}
\begin{itemize}
\item{Measuring blood pressure before and after treatment for 30 patients}
\end{itemize}
\includegraphics[height=5cm]{images/bloodPressure.png}
\end{frame}

\begin{frame}{Mesurements: Dependent / Independent?}
\begin{itemize}
\item{Measuring gene expression in each cell sample, which may be one of five cell-types from cancer patients of normal individuals}
\end{itemize}
\includegraphics[height=5cm]{images/complexData.png}
\end{frame}

\begin{frame}[fragile]{Continuous Data - Distribution}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=10,warning=FALSE,message=FALSE>>=
library(ggplot2)
library(gridExtra)
x     = rnorm(10000, 0, 1)
p1 <- qplot(x,geom="histogram")   + xlim(-4,4) 
p2 <- qplot(x, geom="density")+ xlim(-4,4)
p3 <- qplot(1,x ,geom="boxplot")
grid.arrange(p1,p2,p3,ncol=3)
@

\end{frame}

\begin{frame}[fragile]{Continuous Data - Distribution}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=10,warning=FALSE,message=FALSE>>=
library(RColorBrewer)
x     = rnorm(10000, 0, 1)
sd <- sd(x)
me <- mean(x)

dat <- data.frame(x=x)
rects <- data.frame(xstart = c(me-3*sd,me-2*sd,me-sd), xend = c(me+3*sd,me+2*sd,me+sd),col=letters[1:3])

normdist <- ggplot() +   geom_histogram(data = dat, aes(x))+  geom_rect(data = rects, aes(xmin = xstart, xmax = xend, ymin = -Inf, ymax = Inf,fill=col),alpha=0.4) + ylab("") + xlab("") + geom_segment(aes(x = me-sd, y = 999, xend = me +sd, yend=999),arrow=arrow(length=unit(0.5, "cm"), ends="both")) + geom_segment(aes(x = me-2*sd, y = 979, xend = me +2*sd, yend=979),arrow=arrow(length=unit(0.5, "cm"), ends="both"))  + 
geom_segment(aes(x = me-3*sd, y = 959, xend = me +3*sd, yend=959),arrow=arrow(length=unit(0.5, "cm"), ends="both")) + theme(legend.position="none")
            
normdist
@

\end{frame}

\begin{frame}[fragile]{Continuous or not??}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=10,warning=FALSE,message=FALSE>>=
library(reshape2)
x <- rnorm(200, 0, 2)
y <- rnorm(200, 0, 2)
dataset <- melt(data.frame(x,y))
p1 <- ggplot(dataset, aes(x= value)) + geom_histogram() + facet_wrap(~variable)

x <- rnorm(200, 0, 1)
y <- rnorm(200, 2, 2)
dataset <- melt(data.frame(x,y))
p2 <- ggplot(dataset, aes(x=variable,y= value)) + geom_boxplot()

x <- rnorm(2000, 950, 10)
x <- x[x>950]
dataset <- data.frame(x=x)
p3 <- ggplot(dataset,aes(x)) + geom_histogram()
skewdist <- p3
x <- rnorm(20, 950, 10)
dataset <- data.frame(x=x)

p4 <- ggplot(dataset,aes(x)) + geom_histogram()
grid.arrange(p1,p3,p2,p4,ncol=2)

@

\end{frame}

\begin{frame}[fragile]{Continuous Data - Descriptive Statistics}
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
normdist
@

\begin{itemize}
\item{Measures of location and spread}
\item{Mean and standard deviation}
\begin{itemize}

\item{$\bar{X} = \frac{X_1 + X_2 + \dots X_n}{n}$}
\item{$s.d = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \overline{x})^2}$}
\end{itemize}
\end{itemize}


\end{frame}

\begin{frame}[fragile]{Continuous Data - Descriptive Statistics}
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
skewdist
@
\begin{itemize}
\item{Median: middle value}
\item{Lower quartile: median bottom half of data}
\item{Upper quartile: median top half of data}
\end{itemize}
\end{frame}


\begin{frame}{Continuous Data - Descriptive Statistics Example}

E.g. No. of facebook friends for 7 colleagues\\
311, 345, 270, 310, 243, 5300, 11
\begin{itemize}
\item{Measures of location and spread}
\item{Mean and standard deviation}
\begin{itemize}

\item{$\bar{X} = \frac{X_1 + X_2 + \dots X_n}{n} = 970$}
\item{$s.d = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \overline{x})^2}=1912.57$}

\end{itemize}
\item{Median and interquartile range}
\begin{itemize}
\item{11, {\color{orange}243}, 270, {\color{red}310}, 311, {\color{orange}345},5300}
\end{itemize}
\end{itemize}
\pause
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
friends <- c(311, 345, 270, 310, 243, 5300, 11)
boxplot(friends,horizontal=T)
@

\end{frame}

\begin{frame}{Continuous Data - Descriptive Statistics Example}

E.g. No. of facebook friends for 7 colleagues\\
311, 345, 270, 310, 243, {\color{red}530}, 11


\begin{itemize}
\item{Measures of location and spread}
\item{Mean and standard deviation}
\begin{itemize}

\item{$\bar{X} = \frac{X_1 + X_2 + \dots X_n}{n} = 289$}
\item{$s.d = \sqrt{\frac{1}{N-1} \sum_{i=1}^N (x_i - \overline{x})^2}=153.79$}

\end{itemize}
\item{Median and interquartile range}
\begin{itemize}
\item{11, {\color{orange}243}, 270, {\color{red}310}, 311, {\color{orange}345},530}
\end{itemize}
\end{itemize}
\pause
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
friends <- c(311, 345, 270, 310, 243, 530, 11)
boxplot(friends,horizontal=T)
@
\end{frame}

\begin{frame}{Categorical Data}

\begin{itemize}
\item{Summarised by counts and percentages}
\item{Examples}
\begin{itemize}
\item{19 / 82 (23$\%$) subjects had Grade IV tumour}
\item{48 / 82 (58$\%$) subjects had Diarrheoa as a Adverse Event}
\end{itemize}
\end{itemize}
\end{frame}


\begin{frame}{Standard Deviation and Standard Error}
\begin{itemize}
\item{Commonly confused}
\item{Standard deviation}
\begin{itemize}
\item{Measure of spread of data}
\item{Used for describing \textbf{\textit{population}}}
\end{itemize}
\item{Standard error}
\begin{itemize}
\item{Variability of the mean from repeated sampling}
\item{Precision of the mean}
\item{Used to calculate confidence interval}
\end{itemize}
\item{SD: How widely scattered measurements are}
\item{SE: Uncertainty in \textbf{\textit{estimate}} of sample mean}
\end{itemize}
\end{frame}


\begin{frame}{Confidence intervals for the mean}
\begin{itemize}
\item{Confidence Interval (CI) is a random interval}
\item{In repeated experiments}
\begin{itemize}
\item{95$\%$ of time cover the mean}
\end{itemize}
\item{Looser interpretation 95$\%$ of time}
95$\%$ CI: $(\bar{X} - 1.96 \times se, \bar{X} -+ 1.96 \times se)$\\
\end{itemize}
For facebook friends data:\\
$se = \frac{sd}{\sqrt(n)} = \frac{154}{\sqrt{7}} = 58 $

Mean 289, 95$\%$ CI ( $289 - (1.96 \times 58),  289 + (1.96 \times 58)$)\\
Mean 289, 95$\%$ CI ( 175,  402)

\end{frame}

\begin{frame}[fragile]{Confidence intervals}

\begin{itemize}
\item{As number of observations goes up....}
\item{Standard deviation stays \textbf{the same}...}
\item{But standard error \textbf{goes down....}}
\end{itemize}
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
x <- rnorm(500, 15,1)
y <- sample(x,100)
p1 <- qplot(x, geom="histogram",binwidth=1)
p2 <- qplot(y, geom="histogram",binwidth=1)
grid.arrange(p1,p2,ncol=2)
@

\end{frame}

\section{Hypothesis Testing}
\begin{frame}[fragile]{Basic set-up}
\begin{itemize}
\item{Formulate a \textit{null hypothesis} - H$_0$}
\item{e.g. \textit{"The difference in treatment before and after treatment = 0"}}
\item{Calculate a test statistic from the data under the null hypothesis}
\item{Determine whether the test statistic is more extreme than expected under the null hypothesis}
\item{Reject or do no reject the null hypothesis}
\item{Absence of evidence is not evidence of absense (Bland and Altman, 1995)}
\end{itemize}
\end{frame}

\begin{frame}{Example}
Lady Tasting Tea (Randomised experiment by Fisher)
\begin{itemize}
\item{Randomly-ordered 8 cups of tea}
\begin{itemize}
\item{4 were prepared by adding milk first}
\item{4 were prepared by adding tea first}
\end{itemize}
\item{Task: Lady had to select the 4 cups of one particular method}
\item{H$_0$ Lady no such ability}
\item{Test Statistic: number of sucessses in selecting 4 cups}
\item{Result: Lady got all 4 cups correct}
\item{\textbf{Reject the null hypothesis}}
\end{itemize}

\end{frame}

\begin{frame}[fragile]{Errors}
\includegraphics[height=5cm]{images/errorTable.png}
Significance level, sample size, different of interest, variability of the observations

\textbf{Be aware of issues of multiple testing}
\end{frame}

\section{Tests for continuous variables: T-tests}

\begin{frame}{Various flavours}
\begin{itemize}
\item{One-sample t-test: e.g. H$_0$: mean= 5}
\item{Independent two-sample t-test: e.g. H$_0$: mean of sample 1 = mean of sample 2}
\item{Paired two-sample t-test: e.g. H$_0$: mean difference between pairs = 0 }
\end{itemize}
\end{frame}


\begin{frame}[fragile]{T-distributions}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=

vals <- data.frame(x = rnorm(100000), Distribution = "normal")

vals <- rbind(vals,do.call("rbind",lapply(c(1,3,8,30), function(df) data.frame(x=rt(100000,df=df),Distribution=paste0("df=",df)))))

ggplot(vals, aes(x=x, col=Distribution)) +geom_density()+xlim(-4,4)
@



\end{frame}
\subsection{One-sample t-test}
\begin{frame}{Does mean = X?}

\begin{itemize}
\item{\textbf{Research question:} Published data suggests that the microarray failure rate for a particular supplier is 2.1\%}
\item{Genomics want to know if this holds true in their own lab}
\end{itemize}
\end{frame}

\begin{frame}{Formulating the question}
\begin{itemize}
\item{Null hypothesis, $H_0$}
Mean monthly failure rate $= 2.1\%$
\item{Alternative hypothesis, $H_1$}
Mean monthly failure rate $\ne 2.1\%$
\item{Tails: two-tailed}
\item{Either \textit{reject} or \textit{do not reject} the null hypothesis - \textbf{never accept the alternative hypothesis} }
\end{itemize}
\end{frame}

\begin{frame}[fragile]{The Data}

<<results='asis',echo=FALSE>>=
library(xtable)

failure <- data.frame(Month = month.name, "Monthly failure rate" = c(2.9,2.99,2.48,1.48,2.71,4.17,3.74,3.04,1.23,2.72,3.23,3.4))

xtable(failure)
me <- round(mean(failure$Monthly.failure.rate),3)
sd <- round(sd(failure$Monthly.failure.rate),3)
@

\end{frame}

\begin{frame}[fragile]{Summary Statistics}

\textbf{mean} = $(2.9 + \dots + 3.40) / 12 = \Sexpr{me}$

\textbf{Standard deviation} = \Sexpr{sd}
\begin{itemize}
\item{Observations must be independent}
\item{Observations must be normally distributed}
\end{itemize}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
hist(failure$Monthly.failure.rate,col="steelblue",xlab="Monthly Failure Rate",main="")
@

\end{frame}

\begin{frame}[fragile]{Results}

<<echo=FALSE,results='hide'>>=
test <- t.test(failure$Monthly.failure.rate,mu=2.1)
stat <- round(test$statistic,3)
pval <- round(test$p.value,3)
@

Test statistic
$t_{n-1} = t_{11} = \frac{\bar{x} - \mu_0} {s.d. / \sqrt{n}} = \frac{2.84 - 2.10}{s.e.(\bar{x})} = \Sexpr{stat}$
\pause
<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
library(Rcmdr)
df <- nrow(failure) - 1
.x <- seq(-4.437, 4.437, length.out=1000)
color <- rgb(190, 190, 190, alpha=80, maxColorValue=255)
plotDistr(.x, dt(.x, df=df), cdf=FALSE, xlab="x", ylab="Density", 
main=paste("t Distribution:  Degrees of freedom=11"))

rect(xleft = qt(0.975,df), xright = 5,ytop = 1,ybottom = 0,col=color)
rect(xright = -qt(0.975,df), xleft = -5,ytop = 1,ybottom = 0,col=color)
abline(v = stat,col="red")
@
\pause
\textit{p:value = \Sexpr{pval}}
Reject H$_0$ and conclude that mean monthly failure rate in Genomics is not 2.1\%

\end{frame}

\subsection{Two-sample t-test}

\begin{frame}{Two-sample t-test}
\begin{itemize}
\item{Independent:}
e.g. the weight of two different breeds of mice
\item{Paired:}
e.g. a measurement of disease at two different parts of the body in the same patient
\end{itemize}
\end{frame}


\subsubsection{Independent two-sample t-test}

\begin{frame}{Independent two-sample t-test}
Does mean of group A = mean of group B?


e.g. \textbf{Research question}: 40 male mice (20 of breed A and 20 of breed B) were weighed at 4 weeks old.

\textit{Does the weight of 4 week old male mice depend on breed?}

\end{frame}


\begin{frame}{Independent two-sample t-test}

\begin{itemize}
\item{Null hypothesis, H$_0$}
Mean weight of breed A = Mean weight of breed B
\item{Alternative hypothesis, H$_1$}
Mean weight of breed A $\ne$ Mean weight of breed B
\item{Tails: two-tailed}
\item{Either \textit{reject} or \textit{do not reject} the null hypothesis - \textbf{never accept the alternative hypothesis} }
\end{itemize}
\end{frame}


\begin{frame}[fragile]{The data}

<<results='asis',echo=FALSE>>=
library(xtable)

mice <- data.frame(BreedAMouse = 1:20, WeightA=c(22.77,9.08,9.8,8.13,16.54,11.36,11.47,22.25,14.04,17.12,6.32,17.51,9.87,12.41,7.39,9.23,4.06,8.26,10.24,14.64), BreedBMouse=21:40, WeightB=c(15.51,12.93,11.50,16.07,15.51,15.16,11.25,13.65,14.28,13.21,10.28,12.41,9.63,14.75,12.56,13.02,12.33,11.90,12.57,11.29))

xtable(mice)
@

\end{frame}


\begin{frame}[fragile]{Data summary}

\textbf{Mean} of breed A: \Sexpr{round(mean(mice$WeightA),2)}
\textbf{Mean} of breed B: \Sexpr{round(mean(mice$WeightB),2)}

\textbf{Standard Deviation} of breed A: \Sexpr{round(sd(mice$WeightA),2)}

\textbf{Standard Deviation} of breed B: \Sexpr{round(sd(mice$WeightB),2)}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
df <- data.frame(Breed = c(rep("A", 20), rep("B",20)), Weight = c(mice$WeightA,mice$WeightB))
ggplot(df, aes(x = Weight,fill=Breed)) + geom_histogram() + facet_wrap(~Breed)


@

\end{frame}

\begin{frame}[fragile]{Checking assumptions}

\begin{itemize}
\item{Observations are independent}
\item{Observations are normally distributed}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
df <- data.frame(Breed = c(rep("A", 20), rep("B",20)), Weight = c(mice$WeightA,mice$WeightB))
ggplot(df, aes(x = Weight,fill=Breed)) + geom_histogram() + facet_wrap(~Breed)
test <- t.test(Weight~Breed,data=df,var.equal = FALSE)

tsat <-test$statistic

@
\pause
\item{Use Welch's correction if variances are different ( alters the t-statistic and degrees of freedom)}
\end{itemize}

\end{frame}

\begin{frame}[fragile]{Test results}

$t_{df} = \frac{\bar{X_A} - \bar{X_B}}{s.e.(\bar{X_A} - \bar{X_B})} = \Sexpr{tsat}$

df = 23 (with Welch's correction)

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=



.x <- seq(-3.768, 3.768, length.out=1000)
 plotDistr(.x, dt(.x, df=23), cdf=FALSE, xlab="x", ylab="Density", 
           main=paste("t Distribution:  Degrees of freedom=23"))
color <- rgb(190, 190, 190, alpha=80, maxColorValue=255)
rect(xleft = qt(0.975,23), xright = 5,ytop = 1,ybottom = 0,col=color)
rect(xright = -qt(0.975,23), xleft = -5,ytop = 1,ybottom = 0,col=color)
abline(v = tsat,col="red")

@
\pause
\textit{p-value: \Sexpr{test$p.value}}
\pause
\textbf{Do not reject H$_0$. There is no evidence for a difference in weight between breeds}

\end{frame}

\subsubsection{Paired two-sample t-test}

\begin{frame}{Paired two-sample t-test}
\textit{Does the mean difference = 0?}

\textbf{Research question:} 20 patients with ovarian cancer were studied using MRI imaging. Cellularity was measured for each patient at two sites of disease.

\textit{Does the cellularity differ between the two different sites of disease}

\end{frame}

\begin{frame}
\begin{itemize}
\item{Null hypothesis, H$_0$}
Cellularity at site A = Cellularity at site B
\item{Alternative hypothesis, H$_1$}
Cellularity at site A $\ne$ Cellularity at site B
\item{Tails: two-tailed}
\item{Either \textit{reject} or \textit{do not reject} the null hypothesis - \textbf{never accept the alternative hypothesis} }
\end{itemize}
\end{frame}

\begin{frame}{Null hypothesis}

\begin{itemize}
\item{H$_0$ Cellularity at site A = Cellularity at site B}

\textbf{OR}

\item{H$_0$ Cellularity at site A - Cellularity at site B = 0}

\end{itemize}

\end{frame}

\begin{frame}[fragile]{The data}

<<results='asis',echo=FALSE>>=
cellularity <- read.csv("data/Paired two-sample t-test.csv")
cellularity$Difference = cellularity$Ovarian - cellularity$Peritoneal
xtable(cellularity)

@

\end{frame}

\begin{frame}
Mean difference: \Sexpr{round(mean(cellularity$Difference),3)}
\begin{itemize}
\item{Observations are independent}
\item{The paired differences are normally distributed}
\end{itemize}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=
hist(cellularity$Difference,col="steelblue")
@

\end{frame}

\begin{frame}[fragile]{Test results}

<<echo=FALSE,results='hide'>>=
test <- t.test(cellularity$Difference)

tsat <- round(as.numeric(test$statistic),4)
@

$t_{n-1} = t_{19} = \frac{\bar{X_{A-B}}}{s.e.(\bar{X_{A-B}})} = \Sexpr{tsat}$

df = \Sexpr{nrow(cellularity)-1}

<<echo=FALSE,fig.show='asis',fig.height=5,fig.width=12,warning=FALSE,message=FALSE>>=

df = nrow(cellularity) - 1

  .x <- seq(-3.768, 3.768, length.out=1000)
 plotDistr(.x, dt(.x, df=df), cdf=FALSE, xlab="x", ylab="Density", 
           main=paste("t Distribution:  Degrees of freedom=23"))
color <- rgb(190, 190, 190, alpha=80, maxColorValue=255)
rect(xleft = qt(0.975,df), xright = 5,ytop = 1,ybottom = 0,col=color)
rect(xright = -qt(0.975,df), xleft = -5,ytop = 1,ybottom = 0,col=color)
abline(v = tsat,col="red")

@
\pause
\textit{p-value: \Sexpr{test$p.value}}
\pause
\textbf{Reject H$_0$}. There is evidence that cellularity at site A $\ne$ cellularity at site B


\end{frame}

\begin{frame}{What if normality is not reasonable?}
\begin{itemize}
\item{Transform your data. e.g. log transformation}
\item{Non parametric tests:}
\begin{itemize}
\item{One-sample test $\rightarrow$ One-sample Wilcoxon signed rank test}
\item{Independent two-sample t-test $\rightarrow$ Mann-Whitney U test / Wilcoxon rank sum test}
\item{Paired two-sample t-test $\rightarrow$ Matched-pairs Wilcoxon signed rank test}
\end{itemize}
\end{itemize}

\end{frame}

\begin{frame}{Summary - Continuous variables}
\begin{itemize}
\item{One-sample t-test}
Use when we have \textit{one group}
\item{Independent two-sample t-test}
Use when we have two independent groups. A Welch correction may be needed if the two groups have different spread.
\item{Paired two-sample t-test}
Use when we have \textit{two non-independent groups}
\item{Non-parametric tests or transformations}
Use when we \textit{cannot assume normality}
\end{itemize}

\end{frame}


\end{document}